dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java

```diff
@@ -0,0 +1,543 @@
+
+        try {
+
+            Logger.debug(this, "Finding the workflow action " + actionId);
+            action    = this.workflowAPI.findAction(actionId, user);
+
+            response  = (null == action)?
+                    this.responseUtil.getErrorResponse(request, Response.Status.NOT_FOUND, locale, user.getUserId(), "Workflow-does-not-exists-action"):
+                    Response.ok(new ResponseEntityView(action)).build(); // 200
+        } catch (DotSecurityException e) {
+
+            Logger.error(this.getClass(),
+                    "DotSecurityException on findActionByStep, actionId: " + actionId +
+                            ", exception message: " + e.getMessage(), e);
+            response = ExceptionMapperUtil.createResponse(e, Response.Status.UNAUTHORIZED);
+        } catch (Exception e) {
+
+            Logger.error(this.getClass(),
+                    "Exception on findActionByStep, actionId: " + actionId +
+                            ", exception message: " + e.getMessage(), e);
+            response = ExceptionMapperUtil.createResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+
+        return response;
+    } // findAction.
+
+    /**
+     * Returns a single action associated to the step, 404 if does not exists. 401 if the user does not have permission.
+     * @param request  HttpServletRequest
+     * @param actionId String
+     * @param stepId String
+     * @return Response
+     */
+    @GET
+    @Path("/steps/{stepId}/actions/{actionId}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response findActionByStep(@Context final HttpServletRequest request,
+                                              @PathParam("actionId") final String actionId,
+                                              @PathParam("stepId")   final String stepId) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (null, true, request, true, null);
+        Response response;
+        WorkflowAction action;
+        final Locale locale = LocaleUtil.getLocale(request);
+        final User   user   = initDataObject.getUser();
+
+        try {
+
+            Logger.debug(this, "Getting the workflow action " + actionId + " for the step: " + stepId);
+            action    = this.workflowAPI.findAction(actionId, stepId, user);
+
+            response  = (null == action)?
+                    this.responseUtil.getErrorResponse(request, Response.Status.NOT_FOUND, locale, user.getUserId(), "Workflow-does-not-exists-action"):
+                    Response.ok(new ResponseEntityView(action)).build(); // 200
+        } catch (DotSecurityException e) {
+
+            Logger.error(this.getClass(),
+                    "DotSecurityException on findActionByStep, actionId: " + actionId +
+                            ", stepId: " + stepId + ", exception message: " + e.getMessage(), e);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal ", stepId: " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)